test/integration/otel_test.go

```diff
@@ -0,0 +1,295 @@
+//go:build integration
+
+package integration
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/eggsampler/acme/v3"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/propagation"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/letsencrypt/boulder/cmd"
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/test"
+)
+
+type SpanData struct {
+	SpanID        string
+	OperationName string
+	Warnings      []string
+	ProcessID     string
+	References    []struct {
+		RefType string
+		TraceID string
+		SpanID  string
+	}
+}
+
+type TraceData struct {
+	TraceID   string
+	Spans     []SpanData
+	Processes map[string]struct {
+		ServiceName string
+	}
+	Warnings []string
+}
+
+// TraceResponse is what we get from the traces API.
+// We always search for a single trace by ID, so this should be length 1.
```

Comment:
nit: either add similar comments to SpanData and TraceData, or move those type definitions below this one so that a reader already has context from this comment. (Or both!)

Also, it would be useful to link to the jaeger/otel docs which document this json schema, or which implement it on the response-generation side.

Reply:
Sounds good.  I Will include a link to https://pkg.go.dev/github.com/jaegertracing/jaeger/model/json aka
https://github.com/jaegertracing/jaeger/blob/main/model/json/model.go

Reply:
Alternatively we could just use that package as a dependency, though that's a lot to drag in for a single json object in a unit test.

Reply:
Yeah, no need to take it as a dependency imo; we duplicate structs like this in a few different places. It's just really useful to know exactly what structure is being deserialized into the struct.

Reply:
Reordered, added more comments including references to jaeger's code